File: C++/First Steps in Coding - Basics of C++/While Loops/Graduation.cpp
#include <cstdlib>
#include <iostream>
#include <string>
#include <iomanip>
using namespace std;
int main() {
	string studentName;
	int terminated = 0, k=1;
	double gradesSum = 0;
	getline(cin, studentName);
	while (k <= 12) {
		double grade;
		cin >> grade;
		if (grade < 4.00) {
			terminated += 1;
			if (terminated > 1) {
				cout << studentName << " has been excluded at " << k << " grade";
				break;
			}
		}
		else {
			gradesSum += grade;
			if (k == 12) {
				cout << fixed << setprecision(2) << studentName << " graduated. Average grade: " << gradesSum / 12;
			}
			k++;
		}
	}
}

File: Python/Basics/While Loop/Lab/Graduation.py
studentName = input()
gradeClass = 1
totalGrade = 0.00
fail = 0

while gradeClass <= 12:
    grade = float(input())

    if grade >= 4.00:
        gradeClass += 1
        totalGrade += grade
    else:
        fail += 1

        if fail > 1:
            print(f"{studentName} has been excluded at {gradeClass} grade")
            break

    if gradeClass == 13:
        print(f"{studentName} graduated. Average grade: {totalGrade / 12:.2f}")
        break